src/game/Utils/Cinematics.cc

```diff
@@ -7,200 +7,313 @@
-#include "Types.h"
-#include "VSurface.h"
-#include "HImage.h"
-#include "Video.h"
-#include "UILayout.h"
-
-#include "ContentManager.h"
-#include "GameInstance.h"
-#include "Logger.h"
+#include <cmath>
+#include <vector>
+
+#include <SDL.h>
+extern "C" {
+#include "smacker.h"
+}
+
+#include "externalized/ContentManager.h"
+#include "externalized/GameInstance.h"
+#include "game/Utils/Cinematics.h"
+#include "sgp/Debug.h"
+#include "sgp/FileMan.h"
+#include "sgp/HImage.h"
+#include "sgp/VObject.h"
+#include "sgp/VSurface.h"
+#include "sgp/SoundMan.h"
 
 struct SMKFLIC
 {
-	HWFILE hFileHandle;
-	Smack  *SmackerObject;
-	CHAR8  SmackerStatus;
-	SDL_Surface *SmackBuffer;
-	UINT32 uiFlags;
-	UINT32 uiLeft;
-	UINT32 uiTop;
+	unsigned char* file_in_memory;
+	smk smacker; // object pointer type for libsmacker
+	UINT32 sounds[7];
+	UINT32 flags;
+	UINT32 left;
+	UINT32 top;
+	UINT32 start_tick;
+	UINT32 frame_no;
+	double milliseconds_per_frame;
+	char status;
 };
 
 
-// SMKFLIC uiFlags
+// SMKFLIC flags
 #define SMK_FLIC_OPEN      0x00000001 // Flic is open
 #define SMK_FLIC_PLAYING   0x00000002 // Flic is playing
 #define SMK_FLIC_AUTOCLOSE 0x00000008 // Close when done
 
-static SMKFLIC SmkList[4];
-static UINT32  guiSmackPixelFormat = SMACKBUFFER565;
-
 
-BOOLEAN SmkPollFlics(void)
-{
-	BOOLEAN fFlicStatus = FALSE;
-	FOR_EACH(SMKFLIC, i, SmkList)
-	{
-		if (!(i->uiFlags & SMK_FLIC_PLAYING)) continue;
-		fFlicStatus = TRUE;
+static SMKFLIC gSmkList[4];
 
-		Smack* const smkobj = i->SmackerObject;
 
+static SMKFLIC* SmkOpenFlic(const char* filename);
+static SMKFLIC* SmkGetFreeFlic(void);
+static void SmkSkipFrames(SMKFLIC* sf);
+static void SmkBlitVideoFrame(SMKFLIC* const sf, SGPVSurface* surface);
 
-		if (SmackWait(smkobj)) continue;
 
-		//if (SmackSkipFrames(smkobj)) continue;
+BOOLEAN SmkPollFlics(void)
+{
+	BOOLEAN is_playing = FALSE;
+	FOR_EACH(SMKFLIC, sf, gSmkList)
+	{
+		if (!(sf->flags & SMK_FLIC_PLAYING)) continue;
 
-		{
-			SGPVSurface::Lock l(FRAME_BUFFER);
-			SmackToBuffer(smkobj, i->uiLeft, i->uiTop, l.Pitch(), smkobj->Height, smkobj->Width, l.Buffer<UINT16>(), guiSmackPixelFormat);
-			SmackDoFrame(smkobj);
-		}
+		SmkSkipFrames(sf);
 
-		// Check to see if the flic is done the last frame
-		//printf ("smk->FrameNum %u\n", smk->FrameNum);
-		// if (smk->FrameNum == smk->Frames - 1)
-		if (i->SmackerStatus == SMK_LAST )
+		if (sf->status == SMK_DONE || sf->status == SMK_ERROR)
 		{
-			if (i->uiFlags & SMK_FLIC_AUTOCLOSE) SmkCloseFlic(i);
+			if (sf->flags & SMK_FLIC_AUTOCLOSE)
+			{
+				SmkCloseFlic(sf);
+			}
+			else
+			{
+				sf->status &= ~SMK_FLIC_PLAYING;
+			}
 		}
 		else
 		{
-			i->SmackerStatus = SmackNextFrame(smkobj);
+			is_playing = TRUE;
+			SmkBlitVideoFrame(sf, FRAME_BUFFER);
 		}
 	}
 
-	return fFlicStatus;
+	return is_playing;
 }
 
 
 void SmkInitialize(void)
 {
-	// Wipe the flic list clean
-	memset(SmkList, 0, sizeof(SmkList));
-
-	// Use MMX acceleration, if available
-	SmackUseMMX(1);
```

Comment:
This was removed and not re-added anywhere, so we dont use mmx accelleration anymore?

Reply:
It was function from the radtools dll, it was never used with smacker.